Assets/_BForBoss/_Core/Scripts/Environment/EnemySpawnAreaBehaviour.cs

```diff
@@ -86,18 +93,53 @@ private IEnumerator SpawnEnemies(int count)
         private void SpawnEnemy()
         {
             Logger.LogString("<color=red>Spawn Enemy</color>", "wavesmode");
-            var enemy = _enemyContainer.GetEnemy();
+            Vector3? spawnPosition = GenerateSpawnPosition();
+            if (spawnPosition != null)
+            {
+                var enemy = _enemyContainer.GetEnemy();
             
-            enemy.OnDeath += HandleOnEnemyDeath;
-            enemy.transform.SetPositionAndRotation(transform.position, transform.rotation);
-            enemy.transform.SetParent(transform, true);
+                enemy.OnDeath += HandleOnEnemyDeath;
+                enemy.transform.SetPositionAndRotation((Vector3)spawnPosition, transform.rotation);
+                enemy.transform.SetParent(transform, true);
             
-            //https://answers.unity.com/questions/771908/navmesh-issue-with-spawning-players.html
-            NavMeshAgent agent = enemy.GetComponent<NavMeshAgent>();
-            agent.enabled = false;
-            agent.enabled = true;
+                //https://answers.unity.com/questions/771908/navmesh-issue-with-spawning-players.html
+                NavMeshAgent agent = enemy.GetComponent<NavMeshAgent>();
+                agent.enabled = false;
+                agent.enabled = true;
             
-            _waveModel?.IncrementSpawnCount();
+                _waveModel?.IncrementSpawnCount();
+            }
+        }
+
+        private Vector3? GenerateSpawnPosition()
+        {
+            if (IsPositionValid(transform.position))
+            {
+                return transform.position;
+            }
+
+            const int MaxRetries = 10;
+            for(int tries = 0; tries < MaxRetries; tries++)
+            {
+                var randomPosition = _spawnArea * Random.insideUnitSphere;
+                var randomSpawnPosition = transform.position +
+                                new Vector3(randomPosition.x, 0, randomPosition.z);
+
+                if (NavMesh.FindClosestEdge(randomSpawnPosition, out var hit, NavMesh.AllAreas) && 
+                    IsPositionValid(hit.position))
+                {
+                    return hit.position;
+                }
+            }
+
+            return null;
+        }
+
+        private bool IsPositionValid(Vector3 position)
+        {
+            const int halfExtentSizePositionValidityCheck = 2;
```

Comment:
what is supposed to do exactly?

Reply:
`Physics.OverlapBox()`
https://docs.unity3d.com/ScriptReference/Physics.OverlapBox.html

half extents, for the size of the overlap box

Reply:
fair. personally, I think it would more readable to just do `Vector3 halfExtents = Vector3.one * 2;` but I leave that up to you!

Reply:
To be honest, I would like the actual half extent to be the half extent of the enemy model so that if we change its dimensions or create another enemy to use with completely different proportions, this would be adjusted accordingly. We could technically get that info since we have access to the enemy from `EnemyContainer`, but it could be a bit tedious. What do you think?

Reply:
It's kinda overkill for this.

Reply:
For this task I'm down to keep this as is, but I will probably make a backlog task to make the appropriate change